pennylane/templates/layers/strongly_entangling.py

```diff
@@ -117,9 +117,9 @@ def expand(self):
 
                 for i in range(len(self.wires)):
                     qml.Rot(
-                        self.parameters[0][l, i, 0],
-                        self.parameters[0][l, i, 1],
-                        self.parameters[0][l, i, 2],
+                        self.parameters[0][..., l, i, 0],
+                        self.parameters[0][..., l, i, 1],
+                        self.parameters[0][..., l, i, 2],
```

Comment:
Changes such as this would be required in most of our operation/template classical processing

Reply:
For my understanding: how come we'd need the ellipsis here? Would have thought that the `for p in zip(*params):` syntax in the `batch_dim` transform makes sure that we get a single batch of parameters here.

Reply:
This is because the transform is only called *after* the quantum function is evaluated :) Currently, the transform works as follows:

1. The qfunc is evaluated with the batch dimension QNode parameters
2. A single tape is output, with gates with batch dimension (e.g., `RX([0.1, 0.2, 0.3])`)
3. The transform then loops through the tape, and splits it into multiple circuits, one per batch dimension.

Reply:
@antalszava it's a good idea though, you could write it so that the transform applies _first_